lambdaclass/mirra_backend
# [GH-426] Add Dungeon Mode

> [!CAUTION]
> To be merged with https://github.com/lambdaclass/champions_of_mirra/pull/459

## Motivation

Creates Dungeon Mode for Champions. This PR adds the new campaign as well as the 
Closes https://github.com/lambdaclass/champions_of_mirra/issues/426

## Summary of changes

- Create dungeon mode fight handling (empty apply_buffs function to be filled later)
- Create unit limit in levels
- Add seeds for new super campaign "Dungeon"
- Sends the name of the super campaign instead of its id on SuperCampaignProgress message

## How to test it?

You can't test this with the client since it doesn't have the screen for Dungeon campaign. You can create a user and then check his progress in Dungeon:

```
{:ok, user} = Champions.Users.register "User"
dungeon = GameBackend.Campaigns.get_super_campaign_by_name_and_game("Dungeon", 2)
```

```
GameBackend.Campaigns.get_super_campaign_progress(user.id, dungeon.id)
```

Grab the level_id of the super_campaign_progress and play the level with

```
Champions.Battle.fight_level(user.id, [THE ID])
```

This will advance the super_campaign_progress of the user. Run this and check that the super campaign progress of the "Dungeon" campaign (the one with the id you put in step 2) has changed level id

```
GameBackend.Campaigns.get_super_campaign_progress(user.id, dungeon.id)
```

You can test that locks work by winning the first 10 levels of dungeon mode with a user. You won't be able to win level 11 until you level up your dungeon settlement.
